objectstorage/objectstorage.go

```diff
@@ -0,0 +1,123 @@
+package objectstorage
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/vocdoni/saas-backend/db"
+)
+
+type ObjectStorageConfig struct {
+	DB        *db.MongoStorage
+	ServerURL string
+}
+
+type ObjectStorageClient struct {
+	db        *db.MongoStorage
+	serverURL string
+}
+
+var isObjectNameRgx = regexp.MustCompile(`^([a-zA-Z0-9]+)\.(jpg|jpeg|png)`)
+
+// New initializes a new ObjectStorageClient with the provided API credentials and configuration.
+// It sets up a MinIO client and verifies the existence of the specified bucket.
+func New(conf *ObjectStorageConfig) *ObjectStorageClient {
+	if conf == nil {
+		return nil
+	}
+	return &ObjectStorageClient{
+		db:        conf.DB,
+		serverURL: conf.ServerURL,
+	}
+}
+
+// key is set in a string and can have a directory like notation (for example "folder-path/hello-world.txt")
+func (osc *ObjectStorageClient) Get(objectName string) (*db.Object, error) {
+	if objectName == "" {
+		return nil, fmt.Errorf("objectID is empty")
+	}
+	objectID, ok := objectIDfromName(objectName)
+	if !ok {
+		return nil, fmt.Errorf("invalid objectID")
+	}
+
+	object, err := osc.db.Object(objectID)
+	if err != nil {
+		if err == db.ErrNotFound {
+			return nil, fmt.Errorf("object not found")
+		}
+		return nil, fmt.Errorf("cannot get object: %w", err)
+	}
+
+	return object, nil
+}
+
+// uploadObject uploads the object image with the given objectID, associated to
+// the user with the given userFID and the community with the given communityID.
+// If the objectID is empty, it calculates the objectID from the data. It returns
+// the URL of the uploaded object image. It stores the object in the database.
+// If an error occurs, it returns an empty string and the error.
+func (osc *ObjectStorageClient) Put(data io.Reader, size int64, userID string) (string, error) {
+	// if !isBase64Image(data) {
+	// 	return "", fmt.Errorf("image is not base64 encoded")
+	// }
+
+	// Get the expected size of the data
```

Comment:
to be done?

Reply:
It is passed as a function argument after all